src/theory/strings/inference_manager.cpp

```diff
@@ -113,9 +129,13 @@ void InferenceManager::sendInference(const std::vector<Node>& exp,
                                      bool isRev,
                                      bool asLemma)
 {
-  eq = eq.isNull() ? d_false : Rewriter::rewrite(eq);
-  if (eq == d_true)
+  if (eq.isNull())
+  {
+    eq = d_false;
+  }
+  else if (Rewriter::rewrite(eq) == d_true)
   {
+    // if trivial, return
```

Comment:
I actually think that the negation of this if statement should be an assertion. If we really try to send an inference where the conclusion rewrites to true, we probably created some bogus conclusion.

Reply:
I think there are several places (e.g. inferences via cd simplification) that rely on this as a mechanism for filtering.  If this were an assertion, then this filtering would have to be done in multiple places.